src/metrics.rs

```diff
@@ -3,29 +3,29 @@ use prometheus::{Encoder, IntCounter, IntGauge, Opts};
 
 use crate::LOGGING_PREFIX;
 
-pub type Result<T, E> = std::result::Result<T, E>;
+type Result<T, E> = std::result::Result<T, E>;
```

Comment:
@roshaans please try to use more narrow scope everywhere where it's possible. It's usually private (no keyword) or `pub(crate)`